engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java

```diff
@@ -268,43 +265,16 @@ Trans createInternalTrans() throws KettleException {
     return executorTrans;
   }
 
-  private void passParametersToTrans() throws KettleException {
+  @VisibleForTesting
+  void passParametersToTrans() throws KettleException {
     // Set parameters, when fields are used take the first row in the set.
     TransExecutorParameters parameters = meta.getParameters();
 
     Trans internalTrans = getData().getExecutorTrans();
 
-    internalTrans.clearParameters();
-
-    String[] parameterNames = internalTrans.listParameters();
-    for ( int i = 0; i < parameters.getVariable().length; i++ ) {
-      String variable = parameters.getVariable()[ i ];
-      String fieldName = parameters.getField()[ i ];
-      String inputValue = parameters.getInput()[ i ];
-
-      String value;
-      // Take the value from an input row or from a static value?
-      if ( !Utils.isEmpty( fieldName ) ) {
-        int idx = getInputRowMeta().indexOfValue( fieldName );
```

Comment:
@AliaksandrShuhayeu - could you have a look at PDI-17045, I belive this code change has caused some change in how this step functions. I would appreciate your input. Thank you.

Reply:
@roboguy I answered you in PDI-17045. A problem with the Fields to replace existed  before this changes.

Reply:
Thank you @AliaksandrShuhayeu for the quick reply and taking a look at the ticket! 
As you mentioned in point 2:

"The UI does not populate the Fields to replace (see attached screenshots)"
This problem was not fixed in scope of PDI-16066. And this problem existed before a correction PDI-16066. 

Is there an open issue that deals with the UI problem ? If not would you be able to create it ?

Thank you again for taking a look at this.